Specs/DataSources/PolylineGeometryUpdaterSpec.js

```diff
@@ -561,6 +588,58 @@ defineSuite([
         updater.destroy();
     });
 
+    it('lineType can be dynamic', function() {
+        if (!Entity.supportsPolylinesOnTerrain(scene)) {
+            return;
+        }
+
+        var entity = new Entity();
+        var polyline = new PolylineGraphics();
+        entity.polyline = polyline;
+
+        var time = new JulianDate(0, 0);
+
+        var lineTypeVar = LineType.GEODESIC;
+        var lineType = new CallbackProperty(function() {
+            return lineTypeVar;
+        }, false);
+
+        polyline.show = new ConstantProperty(true);
+        polyline.width = new ConstantProperty(1.0);
+        polyline.positions = new ConstantProperty([Cartesian3.fromDegrees(0, 0, 0), Cartesian3.fromDegrees(0, 1, 0)]);
+        polyline.material = new ColorMaterialProperty(Color.RED);
+        polyline.followSurface = new ConstantProperty(true);
+        polyline.granularity = new ConstantProperty(0.001);
+        polyline.clampToGround = new ConstantProperty(true);
+        polyline.lineType = lineType;
+
+        var updater = new PolylineGeometryUpdater(entity, scene);
+
+        var groundPrimitives = scene.groundPrimitives;
+        expect(groundPrimitives.length).toBe(0);
+
+        var dynamicUpdater = updater.createDynamicUpdater(scene.primitives, groundPrimitives);
+        expect(dynamicUpdater.isDestroyed()).toBe(false);
+        expect(groundPrimitives.length).toBe(0);
+
+        dynamicUpdater.update(time);
+
+        expect(groundPrimitives.length).toBe(1);
+        var primitive = groundPrimitives.get(0);
+
+        expect(primitive.show).toEqual(true);
+
+        lineTypeVar = false;
+        dynamicUpdater.update(time);
+
+        dynamicUpdater.destroy();
+
+        expect(scene.primitives.length).toBe(0);
+        expect(groundPrimitives.length).toBe(0);
```

Comment:
Should changing `LineType` influence whether the underlying `Primitive` is a `Primitive` or `GroundPrimitive`?

Reply:
Maybe have the spec check that the `primitive` has been destroyed and recreated after `lineTypeVar = LineType.STRAIGHT`.